app/models/user.rb

```diff
@@ -65,8 +68,13 @@ class User < ApplicationRecord
   scope :by_skills_in,   ->(*skill_ids) { UsersBySkillsQuery.where(ids: skill_ids) }
   scope :not_in_experience, -> { where arel_table[:created_at].lt(EXPERIENCE_PERIOD.ago) }
   scope :with_level,       -> value { where(level: value) }
+  scope :by_roles_in, lambda { |roles|
+    roles_values = self.roles.find_values(*roles).map(&:value)
```

Comment:
I think you can use `User.roles.values` in here

Reply:
How exactly you suggest? `#find_values` are being used here to find only valid role name